### DISCUSSION TITLE
Alt-[ and Alt-] Do Nothing in PyCharm

### DISCUSSION THREAD

**Discussion by johnballentine** [community/community#7252] [Copilot Conversations]
Hi, I'm on a Windows 10 machine running PyCharm 2021.2.1 Runtime version 11.0.11+9-b1504.16 amd4

When I try to use the 'show next/previous completions' hotkeys, nothing happens.  I have tried it with with several different situations/completions and it never does anything.  It does not, however, write the actual ] or [ characters.  The "Show Completions" Alt-\\ shortcut just flashes the suggested completion for a split second.

I can still see the alternative solutions when going to Tools > GitHub Copilot > Open GitHub Copilot.

I've tried many different key combinations and have the same problem.  Also, there are no other shortcuts occupying alt-[ or alt-]:

![copilot-bug](https://user-images.githubusercontent.com/8368243/140262725-2c94b8f0-cd8a-4198-8bc9-e49353111d79.jpg)

Let me know if there's anything I can do. Thanks!

*EDIT* I still have the same problem 2 months later. I would love to work with anyone on this.

**Comment by jb-0001** [comment]
I experienced the same issue on IntelliJ and WebStorm

**Comment by ICitiesz** [comment]
Hello there, I have the same problem here. Not just the shortcut doesn't work, even the tools menu for GitHub Copilot also grey out for the next & previous suggestion button. I think there might be a bug.

IDE: Intellij IDEA Ultimate 2021.2.3

![image](https://user-images.githubusercontent.com/69842262/140393694-8e275a22-ed28-4f8a-8782-03e6fcd25bb2.png)

![image](https://user-images.githubusercontent.com/69842262/140393550-fc534feb-8b98-4cc2-9473-2eb04d376c59.png)

**Comment by jansorg** [comment]
Thanks for reporting!
Which file type were you editing? Currently, the alternative completions are only available in Python and Java code. But this will be improved in one of the next updates.
`Alt+[` and `Alt+]` seem to be problematic on international keyboards and/or macOS. Could you try to reassign it to something else and try again?  You can assign your own shortcuts using the context menu of the keymap settings.

I didn't fully understand what you meant with
> The "Show Completions" Alt-\ shortcut just flashes the suggested completion for a split second.
Could you elaborate a bit, please?

**Reply by johnballentine** [reply]
I'm using a Windows 10 machine with a US keyboard and Python code.  I spent a while reassigning it and moving around keyboard assignments seeing that it would work, no luck.
There is a shortcut listed in PyCharm's keymap labeled "Show Completion". The default shortcut is Alt-\.
When I hit alt-\, all it does is make the suggested completion (gray code that copilot recommends to autocomplete) disappear while the shortcut is held down, and reappear when the shortcut is released. I'm not sure if the Alt-\ issue is relevant, but I thought I would mention it.  The real problem is that I cannot cycle through completions.

**Comment by seb16120** [comment]
same issue it understood my "alt+]" as alt+) so i put it as the shortcut and it work now

But alt+entrée i still have issue, never worked (even when the blue text box message, which advice you to do this shortcut, is display)

**Comment by gregbrant2** [comment]
Was there ever any progress on this?


I'm new to PyCharm and currently trialling Copilot. I can't get any of the Copilot keyboard shortcuts to work. It's pretty frustrating.
